src/library_pthread.js

```diff
@@ -31,6 +31,28 @@ globalThis.MAX_PTR = (2 ** 32) - 1
 #endif
 }}}
 
+{{{
+globalThis.WORKER_OPTIONS = {
+#if EXPORT_ES6
+  'type': 'module',
+#endif
+#if ENVIRONMENT_MAY_BE_NODE
+  // This is the way that we signal to the node worker that it is hosting
+  // a pthread.
+  'workerData': 'em-pthread',
+#endif
+#if ENVIRONMENT_MAY_BE_WEB || ENVIRONMENT_MAY_BE_WORKER
+  // This is the way that we signal to the Web Worker that it is hosting
+  // a pthread.
+#if ASSERTIONS
+  'name': 'em-pthread-' + PThread.nextWorkerID,
```

Comment:
Alas, the worker options are not entirely static.

Any elegant ideas of how we could solve this? 🤔

Reply:
Vite will let you compile non-static worker options if you include the comment `/* @vite-ignore */` before the worker options definition. It will always treat the worker as a classic worker type if the ignore comment is used.

Perhaps, this would be ok to configure as we only need dynamic name values for debugging?